docs/source/flow-group.rst

```diff
@@ -0,0 +1,107 @@
+.. _flow-group:
+
+==========
+FlowGroups
+==========
+
+This chapter provides information about submitting and running groups of
+multiple operations with varying directives.
+
+.. _flow_group_definition:
+
+Definition
+==========
+
+A :py:class:`FlowGroup` is a collection of one or more operation(s) with the
+directives associated with that operation. An operation can be in multiple
+groups, and the directives for an operation can be different in every group that
+it is in. Practically, a :py:class:`FlowGroup` acts like a "meta-operation" that
+can be submitted and run like an operation. In fact, within the source code,
+each operation is wrapped by a singleton group that handles command generation
+and resource requests for that operation.
+
+.. _flow_group_basic_usage:
+
+Basic Usage
+===========
+
+While a :py:class:`FlowGroup` is automatically created for each operation, users must
+manually create multiple operation groups. Below is an example that creates a
+group named ``ex`` which contains the operations ``op1`` and ``op2``.
+
+.. code-block:: python
+
+    # project.py
+    from flow import FlowProject
+
+    class Project(FlowProject):
+        pass
+
+    ex = Project.make_group(name='ex')
+
+    @ex
+    @Project.operation
+    def op1(job):
+        pass
+
+    @ex
+    @Project.operation
+    def op2(job):
+        pass
+
+    if __name__ == '__main__':
+        Project().main()
+
+To execute or submit only ``ex``, use the option ``--operation`` (``-o``) to
+select the group like you would for a regular operation. A group is eligible if
+at least one of its operations is eligible. Resources (GPUs, MPI ranks, etc.)
+are requested appropriately whether groups are run in series or in parallel.
+
+.. Note::
+
+    Parallel execution of groups means that the cumulative total of operation
+    resources will be requested: e.g. if a group contains two operations ``op1``
+    and ``op2`` and each take two GPUs, then four GPUs will be requested if the
+    group is run in paralel.
+
+    When running the same group in series only 2 GPUs will be requested.
+
+.. tip::
+
+    To avoid wasting resources when submitting groups with multiple operations,
+    make sure that you group operations with similar resource requests or that
+    cheaper operations do not run for long. Mixing GPU operations with highly
+    parallel CPU and will likely leave either the GPU or CPUs idle when the
+    other type of operation is running.
+
+.. _flow-group-specify-directives:
```

Comment:
The `-` is not consistent with the `_` used elsewhere.

Reply:
Good point.